zio/vngio/vectorreader.go

```diff
@@ -86,10 +97,13 @@ func pruneObject(zctx *super.Context, pruner expr.Evaluator, m vng.Metadata) boo
 }
 
 func (v *VectorReader) close() error {
-	if v.activeReaders.Add(-1) <= 0 {
-		if closer, ok := v.readerAt.(io.Closer); ok {
-			return closer.Close() // coffee is for closers
+	var err error
+	v.closeOnce.Do(func() {
+		if v.activeReaders.Add(-1) <= 0 {
```

Comment:
This isn't correct because v.activeReaders.Add can only be called once now.

Reply:
I create a new sync.Once for every ConcurrentPuller and so it will be called once per thread. Am I missing something?

Reply:
Right you are. Missed that.

Reply:
But then it can just be a `bool` since a VectorReader isn't safe to share across goroutines.

Reply:
Right you are I change